aten/src/TH/generic/THTensorMoreMath.cpp

```diff
@@ -28,9 +28,18 @@ int THTensor_(equal)(THTensor *ta, THTensor* tb)
     scalar_t *tbp = tb->data<scalar_t>();
     ptrdiff_t sz = THTensor_(nElement)(ta);
     ptrdiff_t i;
-    for (i=0; i<sz; ++i){
-      if(tap[i] != tbp[i]) return 0;
-    }
+    at::parallel_for(
+        0,
+        sz,
+        HYPER_TH_OMP_OVERHEAD_THRESHOLD,
+        [&](int64_t begin, int64_t end) {
+          for (auto iter = begin; iter < end; iter++) {
+            if (equal && tap[iter] != tbp[iter]) {
```

Comment:
You probably want:
```cpp
if (!equal) { break; }
if (tap[iter] != tbp[iter]) {
	equal = 0;
    break;
}
```

Reply:
Thanks! Updated with your suggestion! The code should have the early exit from the loop when `equal` is 0 already (changed by other threads).